### DISCUSSION TITLE
about gravity

### DISCUSSION THREAD

**Discussion by lo-th** [jrouwe/JoltPhysics.js#129] [Q&A]
Hi i need change gravity of one object in simulation.

is possible on BodyCreationSettings with bcs.mGravityFactor = 0.5;

but after creation of body is not possible to change gravityfactor

**Comment by DennisSmolek** [comment]
You can't do it on the body directly but you can by using the body interface:

https://jrouwe.github.io/JoltPhysics/class_body_interface.html#a2171a36d0ff2b98c019ed4b0bcb0db39

So something like:
`this.bodyInterface.SetGravityFactor(BodyId, 3);`

It's been a hard habit to break for me but I'm learning that with Jolt many methods aren't on the object directly but managed through interfaces/systems/managers.

**Comment by lo-th** [comment]
ok thanks, i see 
but is seems is not exposed in js version

**Comment by jrouwe** [comment]
There's also `MotionProperties::SetGravityFactor` (so `body.GetMotionProperties().SetGravityFactor(...)`). I've exposed both functions in the js version now (but you need to compile it yourself since I didn't create a release).

> It's been a hard habit to break for me but I'm learning that with Jolt many methods aren't on the object directly but managed through interfaces/systems/managers.

Most of the functions on `BodyInterface` are convenience functions that will take a lock on the body and call a function, in this case they just mirror the function on `Body` or `MotionProperties`. Some of them (e.g. setting positions) are only available through the `BodyInterface` as the managing systems also need to be made aware that changes happened.

**Comment by lo-th** [comment]
i have another question
i want apply angularFactor to my body to limite rotation with a vector 
b.GetMotionProperties().SetMaxAngularVelocity( 0 ) work but lock all axis and sometime angular velocity is applied
b.GetMotionProperties().SetAngularVelocityClamped( v ) not work ? 
b.GetMotionProperties().LockAngular( v ) also not work 

any advice ?

ps: in havok they are no angularFactor so i set inertia to [0,0,0]
but inertia in jolt is a mat44

**Reply by lo-th** [reply]
here the [demo ](https://lo-th.github.io/phy/index.html?E=w_jolt#player)
player and 2 back platform should not rotate

**Reply by jrouwe** [reply]
What you want is probably:

```
bodyCreationSettings.mAllowedDOFs = EAllowedDOFs_TranslationX | EAllowedDOFs_TranslationY | EAllowedDOFs_TranslationZ;
```

P.S. Could you next time please create a new discussion for a new issue? That keeps the forum a bit more organized.
